sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -505,6 +496,52 @@ def test_multidimensional_X():
     boost.predict(X)
 
 
+class ClassifierWithoutWeight(DummyClassifier):
+    """Classifier not supporting `sample_weight`."""
```

Comment:
There's a `_NoSampleWeightWrapper` in test_gradient_boosting. It's does something similar. It could be worth to put it in some of the utils modules and reuse it do limit code duplication.